client.go

```diff
@@ -0,0 +1,48 @@
+package id
+
+import (
+	"errors"
+	"github.com/emersion/go-imap"
+	"github.com/emersion/go-imap/client"
+)
+
+// Client is an ID client.
+type Client struct {
+	c *client.Client
+}
+
+// NewClient creates a new client.
+func NewClient(c *client.Client) *Client {
+	return &Client{c: c}
+}
+
+// SupportID checks if the server supports the ID extension.
+func (c *Client) SupportID() (bool, error) {
+	return c.c.Support(Capability)
+}
+
+func (c *Client) id(clientID ID) (serverID ID, err error) {
+	if state := c.c.State(); imap.ConnectedState&state != state {
+		return nil, errors.New("Not connected")
+	}
+
+	var cmd imap.Commander = &Command{ID: clientID}
+
+	res := &Response{}
+	status, err := c.c.Execute(cmd, res)
+	if err != nil {
+		return
+	}
+	if err = status.Err(); err != nil {
+		return
+	}
+
+	serverID = res.ID
+
+	return
+}
+
+// ID sends an ID string to the server and returns the server's ID.
+func (c *Client) ID(clientID ID) (serverID ID, err error) {
+	return c.id(clientID)
```

Comment:
Is there a reason to define a non-exported `id` function if it's only used here?

Reply:
No, there is no explicit reason, I just orientated myself at the [go-imap-move](https://github.com/emersion/go-imap-move) package. I'll fix this.

Reply:
The move package does this because there are two variants of the move command, one with sequence numbers, one with UIDs.

Reply:
Oh, I got it, didn't see that. I've updated the PR.